DOCS: Add docs about CI benchmarking (#16225)
astropy/astropy: docs/development/workflow/maintainer_workflow.rst

  If you wish to run the pre-commit check first in CI without running Actions,
  use ``[skip actions]`` or ``[actions skip]`` in your commit message.

.. _benchmarks:

Benchmarks
==========

If a pull request explicitly deals with a performance improvement, maintainers should
use the ``benchmark`` label to run a comparative benchmark on the pull request. This
will run a benchmark comparing the ``HEAD`` commit of the pull request branch with

> the main branch. The logs are uploaded as a action artefact and can be viewed

Reviewer: ```suggestion
the main branch. The logs are uploaded as a action artifact and can be viewed
```

> in the Actions tab.

Reviewer: ```suggestion
in the Actions tab. This workflow uses the benchmarks
from `Astropy Benchmarks <https://github.com/astropy/astropy-benchmarks/>`_.
```

Maintainers should also run the benchmark if they suspect the code change might have

> performance implications. The benchmark action takes approximately 25 minutes to

Reviewer: ```suggestion
performance implications. The benchmark action takes significantly longer than regular CI to
```

complete.

> It is important to note that the benchmarking on Github Actions will be flakey and

Reviewer: Which one is more correct? 🤷‍♀️ 
```suggestion
It is important to note that the benchmarking on Github Actions will be flaky and
```

Author: No idea. This is taking me to the dark linguistics corners of the internet. It seems like either is fine. But flaky is more used now.
